mlflow/models/evaluation/base.py

```diff
@@ -1716,6 +1722,20 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
             error_code=INVALID_PARAMETER_VALUE,
         )
 
+    # If model_id is specified, verify it matches the derived model_id
+    if specified_model_id is not None and model_id is not None and specified_model_id != model_id:
+        raise MlflowException(
+            message=(
+                f"The specified value of the 'model_id' parameter '{specified_model_id}' "
+                f"contradicts the model_id '{model_id}' associated with the model. Please ensure "
+                f"they match or omit the 'model_id' parameter."
+            ),
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+    # Use specified model_id if provided, otherwise use derived model_id
+    model_id = specified_model_id if specified_model_id is not None else model_id
```

Comment:
```suggestion
    model_id = specified_model_id or model_id
```

Reply:
`is not None` is a safer check than `or`. If `specified_model_id` is a falsy value but not `None`, we don't want to silently fall back to using `model_id` from the pyfunc model; we'd rather throw.

Reply:
Emm that's fair, maybe we should check it's string type

Reply:
I'd propose that the backend should handle that validation

Reply:
Ok that's fine :)